src/dxvk/dxvk_gpu_query.h

```diff
@@ -224,10 +224,12 @@ namespace dxvk {
     uint32_t            m_index;
     bool                m_ended;
 
-    DxvkGpuQueryHandle  m_handle;
-    
-    std::vector<DxvkGpuQueryHandle> m_handles;
-    
+    mutable DxvkQueryData                  m_queryData;
+    mutable DxvkGpuQueryHandle             m_handle;
+    mutable std::deque<DxvkGpuQueryHandle> m_handles;
```

Comment:
I strongly dislike this `mutable` spam; if methods modify internal state then they shouldn't be `const`, especially since doing it like this hides potential thread safety concerns.

The current implementation already has bugs in that regard.

Reply:
I don't really like `mutable` either, though I wanted to avoid a pointer to an extra struct that would have just meant yet one more allocation.

Since `DxvkGpuQuery::getData()` is part of the "public" (D3D layer facing) API, it seemed appropriate to keep it `const` as it is indeed logically const. If that isn't The DXVK Way™, it's easy enough to make `getData()` non-const.

I can replace `m_ended` with an atomic if you'd like, though this is a separate change and I'd rebase this PR to put that commit deeper than this one.